src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
-'use strict';
+"use strict";
+
+const ANIMAL_DAMAGE = 50;
+const ANIMAL_BASE_HP = 100;
+const ANIMAL_MINIMUM_HP = 0;
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = ANIMAL_BASE_HP;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  die() {
+    Animal.alive = Animal.alive.filter((a) => a !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.constructor === Carnivore || animal.hidden) {
```

Comment:
Better to use [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof)